tools/main_light/download_snapshot.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+set -e
+
+GRN='\e[32;1m'
+RED='\033[0;31m'
+BOLDW='\e[1m'
+OFF='\e[0m'
+
+SNAPSHOT_CAR='snapshot.car'
+SNAPSHOT_CHECKSUM='snapshot.sha256sum'
+
+error() {
+  local message="$2"
+  local code="${3:-1}"
+  if [[ -n "$message" ]] ; then
+    echo -e "${RED} Error: ${message}; exiting with status ${code} ${OFF}"
+  else
+    echo -e "${RED} Error; exiting with status ${code} ${OFF}"
+  fi
+  exit_func "${code}"
+}
+
+exit_func() {
+  echo -e "${GRN}Exiting...${OFF}"
+  trap - SIGINT SIGTERM
+  kill -- -$$
+  exit "$1"
+}
+
+download_checksum() {
+  echo -e "${GRN} Checking sha256sum${code} ${OFF}"
+  curl -sI https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car | \
+  perl -ne '/x-amz-website-redirect-location:\s(.+)\.car/ && print "$1.sha256sum"' | xargs wget -O $SNAPSHOT_CHECKSUM
+}
+
+download_snapshot() {
+  echo -e "${GRN} Downloading latest network snapshot ${code} ${OFF}"
+  curl -sI https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car | \
+  perl -ne '/x-amz-website-redirect-location:\s(.+)\.car/ && print "$1.car"' | xargs wget -O $SNAPSHOT_CAR
+}
+
+checksum() {
+  echo "$(cut -c 1-64 $SNAPSHOT_CHECKSUM) $SNAPSHOT_CAR" | sha256sum --check
+}
+
+full_download() {
+  echo -e "${GRN}### Downloading latest snapshot ...${OFF}"
```

Comment:
```
+  cd $SNAPSHOT_DIR
```

Reply:
I guess the `cd` is not necessary if:
```bash
SNAPSHOT_FILE=${SNAPSHOT_DIR}/snapshot.car
SNAPSHOT_CHECK_FILE=${SNAPSHOT_DIR}/snapshot.sha256sum
```

Reply:
Yeah, but  an additional SNAPSHOT_CHECK_FILE seems required to check if it's latest..https://github.com/Zondax/rosetta-filecoin/pull/100/files#diff-6e310cdafcd4921900c679bcbae4dee3584acaff631f55da63e2e47b9d9dec30R59

Reply:
But may be is not required the additional SNAPSHOT_CHECK_FILE and we can go ahead with `SNAPSHOT_CHECK_FILE=${SNAPSHOT_DIR}/snapshot.sha256sum`

Reply:
but it should work now as it is, since  `SNAPSHOT_FILE` and `SNAPSHOT_CHECK_FILE` are absolute path, right ?

Reply:
Nope because this line 
```
    if cmp --silent -- "$SNAPSHOT_CHECKSUM" "../$SNAPSHOT_CHECKSUM"; then
```
It's trying to compare 2  SNAPSHOT_CHECKSUM files... This should be changed to do a real checksum to the car file

Reply:
it only compares the existing checksum `snapshot.sha256sum` with the just downloaded one, just to know if there's a newer snapshot in the s3